dev-docs/howtoguides/troubleshoot_apt_news_security_confinement.md

```diff
@@ -0,0 +1,215 @@
+# How to troubleshoot apt_news security confinement
+
+The `apt-news` service uses two types of security confinements:
+  - [systemd sandboxing features](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#Sandboxing)
+  - [apparmor profile](https://ubuntu.com/server/docs/security-apparmor)
+
+In the git repository, these are located at:
+  - [apt-news.service](https://github.com/canonical/ubuntu-pro-client/blob/main/systemd/apt-news.service)
+  - `debian/apparmor/ubuntu_pro_apt_news.jinja2`
+
+These security features restrict what the service can do on the system, and it's quite common that an application faced with unexpected permission denied errors, or unavailability of resources, will just crash, or behave unexpectedly.
+
+If you suspect the security confinement might be impacting the `apt-news` service, here are some troubleshooting tips.
+
+## Panic: disable everything
+
+To completely remove the security features and make sure they are or are not the cause of the problem you are troubleshooting, do the following:
+
+1. Edit `/lib/systemd/system/apt-news.service` and remove or comment the `AppArmorProfile` line, and the security isolation lines. Here is what the minimal version of that file should look like:
+```
+  [Unit]
+  Description=Update APT News
+
+  [Service]
+  Type=oneshot
+  ExecStart=/usr/bin/python3 /usr/lib/ubuntu-advantage/apt_news.py
+```
+
+An alternative to removing `AppArmorProfile` from the unit file is to just disable it on the system, or put it in complain mode. See "Troubleshooting Apparmor" below for details.
+
+2. Reload the systemd units:
+
+```
+  sudo systemctl daemon-reload
+```
+
+3. Use the service and observe if the problem you are troubleshooting is still there. If it's still there, then the security features are not the cause.
+
+
+
+## Troubleshooting Apparmor
+
+The Apparmor profile for the `apt-news` service is loaded via the `AppArmorProfile` directive in the unit file `/lib/systemd/system/apt-news.service`:
+```
+  [Service]
+  ...
+  AppArmorProfile=ubuntu_pro_apt_news
+```
+
+This will apply the specified apparmor profile on service startup. If the profile does not exist, the service startup will fail. The actual profile is located in `/etc/apparmor.d/ubuntu_pro_apt_news`, and is loaded into the kernel at package install/upgrade time, or when the system boots.
+
+To verify if the Apparmor profile is causing the issues you are observing, the first troubleshooting attempt should be to put it in "complain" mode. In that mode, it will allow everything, but log if something would have been blocked had the profile been in "enforce" mode.
```

Comment:
```suggestion
To verify if the AppArmor profile is causing the issues you are observing, the first troubleshooting attempt should be to put it in "complain" mode. In that mode, it will allow everything, but log if something would have been blocked had the profile been in "enforce" mode.
```